sync_events.py

```diff
@@ -0,0 +1,222 @@
+        self.save()
+
+    def delete(self):
+        self.is_deleted = True
+        self.save()
+
+    @classmethod
+    def load(cls, filepath):
+        with open(filepath) as f:
+            contents = f.read()
+            _, contents = contents.split("---", 1)
+            front_matter, description = contents.split("---", 1)
+
+        front_matter = yaml.load(front_matter, Loader=yaml.FullLoader)
+        description = description.strip()
+
+        if front_matter.get("pretixSlug") is None:
+            # not a pretix event
+            return None
+
+        filename = filepath.name[: -len(filepath.suffix)]
+
+        return cls(
+            filename=filename,
+            pretix_slug=front_matter["pretixSlug"],
+            is_deleted=front_matter["isDeleted"],
+            title=front_matter["title"],
+            event_date=front_matter["eventDate"],
+            event_time=front_matter["eventTime"],
+            event_registration=front_matter["eventRegistration"],
+            correlaidx=front_matter["correlaidx"],
+            languages=front_matter["languages"],
+            tags=front_matter["tags"],
+            description=description,
+        )
+
+    def save(self):
+        front_matter = yaml.dump(
+            {
+                "pretixSlug": self.pretix_slug,
+                "isDeleted": self.is_deleted,
+                "title": self.title,
+                "eventDate": self.event_date,
+                "eventTime": self.event_time,
+                "eventRegistration": self.event_registration,
+                "correlaidx": self.correlaidx,
+                "languages": self.languages,
+                "tags": self.tags,
+            },
+            sort_keys=False,
+        )
+
+        content = f"---\n{front_matter}---\n\n{self.description}"
+
+        filepath = Path(f"content/en/events/{self.event_date[:7]}/{self.filename}.md")
+        filepath.parent.mkdir(parents=True, exist_ok=True)
+        with open(filepath, "w") as f:
+            f.write(content)
+
+    @staticmethod
+    def _parse_title(api_event):
+        # TODO langs
+        return (
+            api_event["name"].get("en")
+            or api_event["name"].get("de")
+            or api_event["name"].get("de-informal")
+        )
+
+    @staticmethod
+    def _parse_date(api_event):
+        date_from = parse_date(api_event["date_from"]).astimezone(pytz.timezone("CET"))
+        return str(date_from.date())
+
+    @staticmethod
+    def _parse_time(api_event):
+        date_from = parse_date(api_event["date_from"]).astimezone(pytz.timezone("CET"))
+        if api_event.get("date_to") is None:
+            return f"{date_from.strftime('%H:%M')} CET"
+        date_to = parse_date(api_event["date_to"]).astimezone(pytz.timezone("CET"))
+        return f"{date_from.strftime('%H:%M')} - {date_to.strftime('%H:%M')} CET"
+
+    @staticmethod
+    def _parse_description(api_event):
+        # TODO
```

Comment:
TODO - best idea so far was web scraping (https://github.com/CorrelAid/hugo-website/issues/196#issuecomment-897435958)... feels a shame the Pretix API doesn't give this data.

Reply:
I still find it hard to believe this information is not somewhere in the API. Had another look and I am wondering what is returned by the "event settings" endpoint:

https://docs.pretix.eu/en/latest/api/resources/events.html#get--api-v1-organizers-(organizer)-events-(event)-settings-

Probably this doesn't have the description, but I'd like to check to be sure. Unfortunately my access token is unable to access that endpoint (403 forbidden). @friep would you be able to give an example of what the "event settings" endpoint returns?

Reply:
we discussed:

* subevents we can use `frontpage_text`
* events we probably need to use the scraping approach again

Reply:
I added the scraping back in for regular events